tutorials/haskell/hspec-golden-test-tutorial/tutorial.md

```diff
@@ -0,0 +1,560 @@
+---
+title: Creating and Updating Golden Test with Hspec-golden
+published: 2019-10-31
+ghc: 8.0.2
+lts: 14.10
+tags: haskell, golden, tests
+libraries: hspec-golden-0.1.0.1
+language: haskell
+author-name: Jorge Andrés Guerra Landázuri
+twitter-profile: Jorge_257
+github-profile: Jagl257
+description: Software testing is necessary to check if an application behaves in a expected way. Sometimes it is useful to store the expected output in a separate file. This is how golden tests works and in this tutorial we will be using Hspec-golden haskell library to create and update with no problem our golden tests.
+---
+
+
+The importance of software testing relies on the ability to check if a determined application behaves in an appropriate way. Unit tests are the first level of testing and helps developers to detect early bugs in the code by testing each component individually.
+
+For example, lets create a function that outputs the string "Hello Golden Testers":
+```haskell
+sayHi :: String
+sayHi = "Hello Golden Testers"
```

Comment:
@Jagl257  I think an example with a little bit more of logic would be more helpful to explain this point. For example:
```hs
fizzBuzz :: [Int] -> [String]
fizzBuzz = ...
```

Then the unit and golden tests will look like:

```hs
spec :: Spec
spec = do
    describe "sayHi" $
        it "shows a Hello Golden Testers string" $
            fizzBuzz [1,2,3,5,15] `shouldBe` ["1", "2", "Fizz", "Buzz", "FizzBuzz"]

spec :: Spec
spec = do
    describe "sayHi" $
        it "shows a Hello Golden Testers string" $
            defaultGolden 'fizzBuzz' (show $ fizzBuzz [1,2,3,5,15])
{- Will store the file:
# .golden/fizzBuzz/golden
" [\"1\", \"2\", \"Fizz\", \"Buzz\", \"FizzBuzz\"]"
-}
```

Reply:
What do you think?

Reply:
That test is just helpful to make an Intro. But I think this example could be used in the very first golden test sample. What do you think?

Reply:
Sounds good @Jagl257 :+1:

Reply:
Its done @CristhianMotoche, To do this i had to chage some things in the tutorial as well.